contracts/Aavegotchi/facets/AavegotchiLendingFacet.sol

```diff
@@ -0,0 +1,331 @@
+        rental.timeAgreed = block.timestamp;
+
+        uint256 tokenId = rental.erc721TokenId;
+        s.lentTokenIdIndexes[originalOwner][tokenId] = s.lentTokenIds[originalOwner].length;
+        s.lentTokenIds[originalOwner].push(tokenId);
+
+        if (rental.erc721TokenAddress == address(this)) {
+            LibAavegotchi.transfer(originalOwner, renter, tokenId);
+        } else {
+            // External contracts
+            IERC721(rental.erc721TokenAddress).safeTransferFrom(originalOwner, renter, tokenId);
+        }
+
+        // set original owner as pet operator
+        s.petOperators[renter][originalOwner] = true;
+
+        emit ERC721ExecutedRental(
+            _rentalId,
+            originalOwner,
+            renter,
+            rental.erc721TokenAddress,
+            tokenId,
+            rental.amountPerDay,
+            rental.period,
+            block.timestamp
+        );
+    }
+
+    ///@notice Allow to claim revenue from the rental
+    ///@dev Will throw if the NFT has not been lent or if the rental has been canceled already
+    ///@param _tokenId The identifier of the lent aavegotchi to claim
+    ///@param _revenueTokens The address array of the revenue tokens to claim; FUD, FOMO, ALPHA, KEK, then GHST
+    function claimAavegotchiRental(uint256 _tokenId, address[] calldata _revenueTokens) external {
+        uint256 rentalId = s.aavegotchiRentalHead[_tokenId];
+        require(rentalId != 0, "AavegotchiLending: rental not found");
+        AavegotchiRental storage rental = s.aavegotchiRentals[rentalId];
+
+        address sender = LibMeta.msgSender();
+        address originalOwner = rental.originalOwner;
+        address renter = rental.renter;
+        address receiver = rental.receiver;
+        uint256 completionTime = rental.timeAgreed + rental.period * 1 days;
+        if (completionTime + 1 days > block.timestamp) {
+            require(originalOwner == sender, "AavegotchiLending: only owner can claim during agreement");
+        }
+
+        uint256 tokenId = rental.erc721TokenId;
+        address escrow = s.aavegotchis[tokenId].escrow;
+        address collateralType = s.aavegotchis[tokenId].collateralType;
+        for (uint256 i; i < _revenueTokens.length; i++) {
+            address revenueToken = _revenueTokens[i];
+            if (escrow == address(0)) continue;
+            if (collateralType == revenueToken) continue;
+
+            uint256 balance = IERC20(revenueToken).balanceOf(escrow);
+            if (balance == 0) continue;
+
+            if (IERC20(revenueToken).allowance(escrow, address(this)) < balance) {
+                CollateralEscrow(escrow).approveAavegotchiDiamond(revenueToken);
+            }
+            if ((completionTime + 1 days <= block.timestamp) && (sender != originalOwner)) {
+                uint256 senderAmount = (balance * 3) / 100;
+                LibERC20.transferFrom(revenueToken, escrow, sender, senderAmount);
+                balance -= senderAmount;
```

Comment:
It is safer,  with regards to a reentry attack to put this line `balance -= senderAmount;`  above the transfer, above line 255.